dcc-portal-ui/app/scripts/survivalanalysis/js/survival-analysis.js

```diff
@@ -0,0 +1,302 @@
+        .attr('class', 'line')
+        .attr('d', line)
+        .attr('stroke', setColor);
+
+      // draw the data points as circles
+      setGroup.selectAll('circle')
+        .data(data.donors)
+        .enter()
+        .append('svg:circle')
+          .attr('class', 'point')
+          .attr('status', function (d) { return d.status; })
+          .attr('cx', function(d) { return x(d.survivalTime); })
+          .attr('cy', function(d) { return y(d.survivalEstimate); })
+          .attr('fill', setColor )
+          .on('mouseover', function (d) {
+            onMouseEnterDonor(d3.event, d);
+          })
+          .on('mouseout', function (d) {
+            onMouseLeaveDonor(d3.event, d);
+          })
+          .on('click', function (d) {
+            onClickDonor(d3.event, d);
+          })
+        ;
+    });
+
+    return svg;
+  }
+
+  var survivalAnalysisController = [
+    '$scope',
+    '$element',
+    function (
+      $scope,
+      $element
+    ) {
+      var ctrl = this;
+      var graphContainer = $element.find('.survival-graph').get(0);
+      var svg = d3.select(graphContainer).append('svg');
+      var tipTemplate = _.template($element.find('.survival-tip-template').html());
+
+      this.disabledDataSets = [];
+
+      var update = function () {
+        if (!ctrl.dataSets) {
+          return;
+        }
+        svg.selectAll('*').remove();
+        renderChart({
+          svg: svg, 
+          container: graphContainer, 
+          dataSets: ctrl.dataSets,
+          disabledDataSets: ctrl.disabledDataSets,
+          palette: ctrl.palette,
+          onMouseEnterDonor: function (event, donor) {
+            $scope.$emit('tooltip::show', {
+              element: event.target,
+              text: tipTemplate({
+                donor: _.extend(
+                  { isCensored: _.includes(ctrl.censoredStatuses,donor.status) },
+                  donor
+                ),
+                labels: ctrl.tipLabels,
+                unit: 'days'
+              }),
+              placement: 'right',
+              sticky: true
+            });
+          },
+          onMouseLeaveDonor: function () {
+            $scope.$emit('tooltip::hide');
+          },
+          onClickDonor: function (e, donor) {
+            window.open('/donors/'+donor.id, '_blank');
+          }
+        });
+      };
+
+      window.addEventListener('resize', update);
+      update();
+
+      this.isDataSetDisabled = function (dataSet) {
+        return _.includes(ctrl.disabledDataSets, dataSet);
+      };
+
+      this.toggleDataSet = function (dataSet) {
+        ctrl.disabledDataSets = _.xor(ctrl.disabledDataSets, [dataSet]);
+        ctrl.isDataSetDisabled(dataSet);
+        update();
+      };
+
+      this.$onChanges = function (changes) {
+        if (changes.dataSets) {
+          setTimeout(update);
+        }
+      };
+
+      this.$onDestroy = function () {
+        window.removeEventListener('resize', update);
+      };
+
```

Comment:
Blank line between function definitions for ease of scanning, is that not typically done here?
We can make an eslinting rule for consistency's sake